pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -246,6 +252,12 @@ func (r *ReconcileCustomDomain) Reconcile(request reconcile.Request) (reconcile.
 	reqLogger.Info(fmt.Sprintf("DNSRecord %s created with value %s", dnsRecordName, dnsRecord.Spec.DNSName))
 	instance.Status.DNSRecord = dnsRecord.Spec.DNSName
 
+	// endpoint is a resolvable dns address w/ a random host under the ingress domain
+	if len(instance.Status.Endpoint) == 0 {
+		endpoint := fmt.Sprintf("%s.%s", randSeq(hostLength), ingressDomain)
```

Comment:
See comment above about moving this into `.Spec` - This specifically if it didn't exist in `.Status` we'd generate a new host name?

Reply:
I think we can leave `Endpoint` in Status as it's really an ephemeral hostname that, in the rare scenario, change.